src/AuthServer.cpp

```diff
@@ -97,6 +105,81 @@ std::string getContent(char const* url)
     return buffer;
 }
 
+std::string getRSAPublicKeyInPEMFormat(std::string_view nnInBase64UrlUnpadded, std::string_view eeInBase64UrlUnpadded)
+{
+    auto nnBin = cppcodec::base64_url_unpadded::decode(nnInBase64UrlUnpadded);
+    auto eeBin = cppcodec::base64_url_unpadded::decode(eeInBase64UrlUnpadded);
+    BIGNUM* modul = BN_bin2bn(nnBin.data(),nnBin.size(),NULL);
+    BIGNUM* expon = BN_bin2bn(eeBin.data(),eeBin.size(),NULL);
+    RSA* rr = RSA_new();
```

Comment:
Should "rr" be freed or ownership was taken by some of the functions?

Reply:
Will investigate, most likely you are right and I should free this somewhere... will think about it. Thank you very much!